src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -373,19 +374,13 @@ public static void WriteAllBytes(string path, byte[] bytes)
             if (bytes == null)
                 throw new ArgumentNullException(nameof(bytes));
 
-            InternalWriteAllBytes(path, bytes);
-        }
-
-        private static void InternalWriteAllBytes(string path, byte[] bytes)
-        {
-            Debug.Assert(path != null);
-            Debug.Assert(path.Length != 0);
-            Debug.Assert(bytes != null);
-
-            using (FileStream fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.Read))
-            {
-                fs.Write(bytes, 0, bytes.Length);
-            }
+#if MS_IO_REDIST
+            using FileStream fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.Read);
+            fs.Write(bytes, 0, bytes.Length);
+#else
+            using SafeFileHandle sfh = OpenHandle(path, FileMode.Create, FileAccess.Write, FileShare.Read);
+            RandomAccess.Write(sfh, bytes, 0);
```

Comment:
I would love to use `RandomAccess` here and even in more places, however, I have one concern: what about paths that point to pipes or other unseekable files?

On Windows we don't allow for opening `FileStream` from a path that points to a non-regular file:

https://github.com/dotnet/runtime/blob/0696727ece4d363f171f80dd442b61fb20c6500a/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs#L145-L154

However, on Unix we just don't allow for paths that point to directories:
https://github.com/dotnet/runtime/blob/0696727ece4d363f171f80dd442b61fb20c6500a/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs#L72

So in theory the following code could work on Unix so far:

```cs
File.WriteAllBytes($pathToApipe, new byte[1]);
```

Now it would throw. Would this be considered as a breaking change?

FWIW I would like to remove the Windows limitation (https://github.com/dotnet/runtime/pull/54676)

Reply:
> Now it would throw.

It shouldn't as of #55191.  We can add a test, though.